### ISSUE TITLE
Sign Up: email address verification email filtered as SPAM 

### ISSUE THREAD

**Issue by joshi1983** [hhaccessibility/hhaccessibility.github.io#275]
When signing up by filling in first name, last name, a password... an email is sent to the specified email address which contains a link which must be clicked before the new account can be used.  The email is getting filtered as SPAM or Trash in gmail and hotmail.

***Steps to reproduce***:
1. Sign up on [app.accesslocator.com/signup](http://app.accesslocator.com/signup) by submitting the form.  Don't use facebook or google authentication.
1. Check your email and the email should be found in your trash or spam folder.

Bruce probably has better ideas of how to fix this but I(Josh) suspect it is either a misconfiguration of MX records in DNS or that the content of the email looks suspicious to SPAM filters since it contains HTML and just short text with a link.

This is related to #199.

**Comment by joshi1983** [comment]
@brucelau-github do you have time to look at this?  This isn't really urgent but would be nice to fix by April 20 or so because that would let us put the email verification in our next demo release.

**Comment by stephanosm** [comment]
Also, I recommend to change the message wording by adding "Click on the link below or" before the word "just"   //see below part of the existing message:
_It's easy — just copy and past link below into your browser._

**Comment by joshi1983** [comment]
@Blandine-AA regarding the "help wanted" label, this issue isn't suitable to most of the student volunteers because it requires some knowledge of DNS, MX records, SMTP, and how email filters work which most students probably lack experience with.  Access to the digital ocean hosting account would also be needed which only @brucelau-github and I are trusted with for now.

We can keep the "help wanted" label on it but just wanted to explain how it was no accident to leave it out and that this is pointing specifically at 1 person for help.

**Comment by joshi1983** [comment]
@brucelau-github are you getting anymore time to work on this lately?

**Comment by brucelau-github** [comment]
Yeah. I would fix this problem quickly.

**Comment by joshi1983** [comment]
@brucelau-github is working on this.  He has this fixed for gmail accounts but ran into some problems with yahoo and hotmail email SPAM filters.  He's still working on this when he can.

**Comment by brucelau-github** [comment]
Received: from PU1APC01HT224.eop-APC01.prod.protection.outlook.com
 (2a01:111:e400:5bcf::50) by KL1PR01MB0727.apcprd01.prod.exchangelabs.com with
 HTTPS via KL1PR02CA0040.APCPRD02.PROD.OUTLOOK.COM; Tue, 9 May 2017 00:34:25
 +0000
Received: from PU1APC01FT043.eop-APC01.prod.protection.outlook.com
 (10.152.252.53) by PU1APC01HT224.eop-APC01.prod.protection.outlook.com
 (10.152.253.226) with Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384_P384) id 15.1.1047.9; Tue, 9 May
 2017 00:34:23 +0000
**Authentication-Results: spf=pass** (sender IP is 159.203.30.1)
 smtp.mailfrom=accesslocator.com; hotmail.com; **dkim=pass** (signature was
 verified) header.d=accesslocator.com;hotmail.com; dmarc=bestguesspass
 action=none header.from=accesslocator.com;
Received-SPF: Pass (protection.outlook.com: domain of accesslocator.com
 designates 159.203.30.1 as permitted sender) receiver=protection.outlook.com;
 client-ip=159.203.30.1; helo= accesslocator.com;
Received: from COL004-MC2F22.hotmail.com (10.152.252.58) by
 PU1APC01FT043.mail.protection.outlook.com (10.152.253.6) with Microsoft SMTP
 Server (version=TLS1_2, cipher=TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384_P384) id
 15.1.1047.9 via Frontend Transport; Tue, 9 May 2017 00:34:21 +0000

**Comment by brucelau-github** [comment]
It seems our server has been past the test in email verification

**Comment by brucelau-github** [comment]
It ready to merge now.

**Comment by joshi1983** [comment]
I removed the "Working on it" label because @brucelau-github was working on other issues lately.  This email SPAM problem still isn't perfect but it probably never will be.  A few improvements were made from Bruce's work.  

Bruce, can you summarize the status on this?  Which email services sent AccessLocator emails to SPAM or Trash as apposed to the Inbox?  Leaving a status here would help yourself or anyone else wanting to continue this work and decide if we want to close this issue.

**Comment by joshi1983** [comment]
The required portion of this for milestone 0.0.5 is:
- verify that the user knows to check spam feature when sending any signup email
- If the user isn't clearly told to check SPAM in email, change things to clearly pass along the message.

The problem of some email getting in SPAM will be very difficult to solve and won't be necessary as part of 0.0.5.  It is likely impossible to guarantee that all email reaches in the Inbox but some improvements could be implemented later.

**Comment by brucelau-github** [comment]
This problem could be solved but using third party email SMTP service. Their IP address and CA trust relationship has been broadcasted across the internet so emails sent over their smtp servers would not be treated as SPAM.

**Comment by joshi1983** [comment]
@brucelau-github do you recommend any specific third party email SMTP service?  Are there any free ones we can use?  Would it be possible to use our accesslocator@gmail.com account somehow to send email for free or would we need to pay for the service?

The HTTPS SSL certificate involved with #77 could use some of your help verifying the most cost-effective solution.

**Comment by brucelau-github** [comment]
There are some options we could choose.  I think mailgun is a good choice to us. It is popular and "free".

Laravel provides a clean, simple API over the popular SwiftMailer library. Laravel provides drivers for SMTP, Mailgun, Mandrill, SparkPost, Amazon SES, PHP's mail function, and sendmail, allowing you to quickly get started sending mail through a local or cloud based service of your choice.
Driver Prerequisites

The API based drivers such as Mailgun and Mandrill are often simpler and faster than SMTP servers. All of the API drivers require that the Guzzle HTTP library be installed for your application. You may install Guzzle to your project by adding the following line to your composer.json file:

"guzzlehttp/guzzle": "~5.3|~6.0"

Mailgun Driver

To use the Mailgun driver, first install Guzzle, then set the driver option in your config/mail.php configuration file to mailgun. Next, verify that your config/services.php configuration file contains the following options:

'mailgun' => [
    'domain' => 'your-mailgun-domain',
    'secret' => 'your-mailgun-key',
],

[laravel mail](https://laravel.com/docs/5.2/mail)

**Comment by brucelau-github** [comment]
I think we dont have too much traffic at all.

https://www.mailgun.com/pricing

**Comment by Blandine-AA** [comment]
PASS
